src/lib/y2partitioner/dialogs/lvm_vg.rb

```diff
@@ -0,0 +1,267 @@
+      # @param controller [Actions::Controllers::LvmVg]
+      def initialize(controller)
+        textdomain "storage"
+        @controller = controller
+      end
+
+      # @macro seeDialog
+      def title
+        _("Add Volume Group")
+      end
+
+      # @macro seeDialog
+      def contents
+        VBox(
+          Left(HVSquash(NameWidget.new(controller))),
+          Left(HVSquash(ExtentSizeWidget.new(controller))),
+          DevicesWidget.new(controller)
+        )
+      end
+
+    private
+
+      # @return [Actions::Controllers::LvmVg]
+      attr_reader :controller
+
+      # Widget for the LVM volume group name
+      class NameWidget < CWM::InputField
+        # Constructor
+        #
+        # @param controller [Actions::Controllers::LvmVg]
+        def initialize(controller)
+          textdomain "storage"
+          @controller = controller
+        end
+
+        # @macro seeAbstractWidget
+        def opt
+          [:notify]
+        end
+
+        # @macro seeAbstractWidget
+        def label
+          # TRANSLATORS: field to enter the name of a new volume group
+          _("&Volume Group Name")
+        end
+
+        # @macro seeAbstractWidget
+        def init
+          self.value = controller.vg_name
+          focus
+        end
+
+        # @macro seeAbstractWidget
+        def handle
+          controller.vg_name = value
+          nil
+        end
+
+        # @macro seeAbstractWidget
+        # Checks whether the given value is valid
+        #
+        # @note An error popup is shown when necessary.
+        #
+        # @return [Boolean]
+        def validate
+          focus
+          presence_validation && legal_characters_validation && uniqueness_validation
+        end
+
+      private
+
+        # @return [Actions::Controllers::LvmVg]
+        attr_reader :controller
+
+        # Sets the focus into this widget
+        def focus
+          Yast::UI.SetFocus(Id(widget_id))
+        end
+
+        # Checks whether some value was entered
+        #
+        # @return [Boolean]
+        def presence_validation
+          return true unless controller.empty_vg_name?
+
+          Yast::Popup.Error(
+            _("Enter a name for the volume group.")
+          )
+
+          false
+        end
+
+        # Checks whether the entered value has only legal characters
+        #
+        # @return [Boolean]
+        def legal_characters_validation
+          return true unless controller.illegal_vg_name?
+
+          Yast::Popup.Error(
+            _("The name for the volume group contains illegal characters. Allowed\n" \
+              "are alphanumeric characters, \".\", \"_\", \"-\" and \"+\"")
+          )
+
+          false
+        end
+
+        # Checks whether there is not another device with the entered name
+        #
+        # @return [Boolean]
+        def uniqueness_validation
+          return true unless controller.duplicated_vg_name?
+
+          Yast::Popup.Error(
+            format(
```

Comment:
Maybe a comment for translators here. My feeling is that they are used to the `%s` and `%1` notation, but not so much to the Ruby keyword placeholders one.

Reply:
Added.